klayout/drc/rule_decks/mim_b.drc

```diff
@@ -0,0 +1,149 @@
+  mimtm5_l1.output('MIMTM.5',
+                   'MIMTM.5 : Minimum spacing between top plate and the Vian-1 connecting to the bottom plate: 0.4µm')
+  mimtm5_l1.forget
+
+  # Rule MIMTM.6: Minimum spacing between unrelated top plates is 0.6µm.
+  logger.info('Executing rule MIMTM.6')
+  mimtm6_l1 = fusetop.space(0.6.um, euclidian)
+  mimtm6_l1.output('MIMTM.6', 'MIMTM.6 : Minimum spacing between unrelated top plates: 0.6µm')
+  mimtm6_l1.forget
+
+  # Rule MIMTM.7: Min FuseTop enclosure by CAP_MK is 0um.
+  logger.info('Executing rule MIMTM.7')
+  mimtm7_l1 = fusetop.not(cap_mk)
+  mimtm7_l1.output('MIMTM.7', 'MIMTM.7 : Min FuseTop enclosure by CAP_MK: 0um')
+  mimtm7_l1.forget
+
+  # Rule MIMTM.8a: Minimum MIM cap area (defined by FuseTop area) is 25µm².
+  logger.info('Executing rule MIMTM.8a')
+  mimtm8a_l1 = fusetop.with_area(nil, 25.um)
+  mimtm8a_l1.output('MIMTM.8a', 'MIMTM.8a : Minimum MIM cap area (defined by FuseTop area): 25µm²')
+  mimtm8a_l1.forget
+
+  # Rule MIMTM.8b: Maximum single MIM Cap area (Use multiple MIM caps in
+  ## parallel connection if bigger capacitors are required) (um2) is 10000µm.
+  logger.info('Executing rule MIMTM.8b')
+  mimtm8b_l1 = fusetop.with_area(10_000.um, nil).not_in(fusetop.with_area(10_000.um))
+  mimtm8b_l1.output('MIMTM.8b', "MIMTM.8b : Maximum single MIM Cap area (Use multiple MIM caps in
+                     parallel connection if bigger capacitors are required) (um2): 10000µm")
+  mimtm8b_l1.forget
+
+  # Rule MIMTM.9: Min. Via (Vian-1) spacing for sea of Via on MIM top plate is 0.5µm.
+  logger.info('Executing rule MIMTM.9')
+  mimtm9_l1 = top_via.inside(fusetop).space(0.5.um, euclidian)
+  mimtm9_l1.output('MIMTM.9', 'MIMTM.9 : Min. Via (Vian-1) spacing for sea of Via on MIM top plate: 0.5µm')
+  mimtm9_l1.forget
+
+  # Rule MIMTM.10: (a) There cannot be any Vian-2 touching MIM bottom plate Metaln-1.
+  ## (b) MIM bottom plate Metaln-1 can only be connected through the higher Via (Vian-1).
+  logger.info('Executing rule MIMTM.10')
+  mimtm10_l1 = topmin1_via.interacting(topmin1_metal_fusetop)
+  mimtm10_l1.output('MIMTM.10', "MIMTM.10 : (a) There cannot be any Vian-2 touching MIM bottom plate Metaln-1.
+                    (b) MIM bottom plate Metaln-1 can only be connected through the higher Via (Vian-1).")
+  mimtm10_l1.forget
+
+  # Rule MIMTM.11: Bottom plate of multiple MIM caps can be shared (for common nodes)
+  ## as long as total MIM area with that single common plate does not exceed MIMTM.8b rule.
+  logger.info('Executing rule MIMTM.11')
+  mimtm11_large_topmin1_metal = topmin1_metal_fusetop.with_area(10_000, nil)
+  mimtm11_large_topmin1_metal_violation = polygon_layer
+  mimtm11_large_topmin1_metal.data.each do |p|
+    mimtm11_topmin1_metal_polygon_layer = polygon_layer
+    mimtm11_topmin1_metal_polygon_layer.data.insert(p)
+    fuse_in_polygon = fusetop.and(mimtm11_topmin1_metal_polygon_layer)
```

Comment:
wouldn't it be more effective to test `fusetop.and(mimtm11_large_topmin1_metal)` outside of a loop, and then look if any resulting shape has an area > 10_000?

Reply:
@proppy We have to use it inside the loop as we need to iterate for each single plate and check its area using this `fuse_in_polygon.area > 10_000`, the operand `.area` returns the total area of the polygons, so if we keep it outside loop, we will get false results.

Reply:
@proppy We have to use all checks inside the loop as we need to iterate over each single plate.

Reply:
yes, but is there a difference between:
```
fuse_in_polygon = fusetop.and(mimtm11_large_topmin1_metal)
fuse_in_polygon_bad = fuse_in_polygon.filter {|p| p > 10_000 }
```
and what we currently have?